ironfish/src/wallet/wallet.ts

```diff
@@ -672,7 +672,7 @@ export class Wallet {
 
   async send(
     memPool: MemPool,
-    sender: Account,
+    sender: SpendingAccount,
```

Comment:
I thought based on the original spec, we were just gonna take a normal account here and do the check in here? This kind of puts a code burden on the external caller to do this check everywhere which balloons out the places needed to do this check. It's ok but it's kind of an irritating SDK design when in reality we have good infrastructure to automatically catch those errors and the user knows what to do with them.

Reply:
Adding this to the typing system is more explicit IMO. Lets consider this `send` example

If someone writes new code and wants to call `send`, they could pass an account (without spending key) and compiler checks would say they are good. It wouldn't be until they wrote a test (which we can only hope would test both types of accounts) or ran the code that they would see there is a failure. We are more explicitly defining our signature and therefore the interface without the user needing to understand the intracicies of what happens via the `send` command. 

The send method does not need an `Account` passed, it _needs_ an account with a spending key.